app/decorators/models/solidus_address_name/address_name_decorator.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module SolidusAddressName
+  module AddressNameDecorator
+    # Creates an instance of Spree::Address::Name parsing input attributes.
+    # @param attributes [Hash] an hash possibly containing name-related
+    #   attributes (name, firstname, lastname, first_name, last_name)
+    # @return [Spree::Address::Name] the object created
+    def self.from_attributes(attributes)
+      params = attributes.with_indifferent_access
+
+      if params[:name].present?
+        Spree::Address::Name.new(params[:name])
+      elsif params[:firstname].present?
+        Spree::Address::Name.new(params[:firstname], params[:lastname])
+      elsif params[:first_name].present?
+        Spree::Address::Name.new(params[:first_name], params[:last_name])
+      else
+        Spree::Address::Name.new
+      end
+    end
+
+    Spree::Address::Name.prepend(self)
```

Comment:
We should add the tests for this method.

Reply:
Why did you create 2 decorators AddressNameDecorator and AddressDecorator?
It should be enough to have 1 decorator with all methods.

Reply:
@mcrisologo why you marked this conversation as resolved without changing anything and without answering?
I'm missing something?